docs/references/function_calling.md

```diff
@@ -0,0 +1,155 @@
+# Tools Calling
+
+Currently, we added the support for tools calling in the following models:
+  - Llama 3.2 models
+  - Llama 3.1 models
+  - Qwen 2.5 models
+  - InternLM Models
+
+For adding support of more different architectures: \n
+ 1. Add Tool tag present in model in line 1055 `sglang/srt/openai_api/adapter.py`
+ 2. Add support in parsing function for converting into tool calls `sglang/srt/utils.py`
+
+
+## Single Round Invocation
+```python
+tools = [
+  {
+    "type": "function",
+    "function": {
+      "name": "get_current_weather",
+      "description": "Get the current weather in a given location",
+      "parameters": {
+        "type": "object",
+        "properties": {
+          "location": {
+            "type": "string",
+            "description": "The city and state, e.g. San Francisco, CA",
+          },
+          "unit": {"type": "string", "enum": ["celsius", "fahrenheit"]},
+        },
+        "required": ["location"],
+      },
+    }
+  }
+]
+
+messages = [{"role": "user", "content": "What's the weather like in Boston today?"}]
+
+client = OpenAI(api_key='YOUR_API_KEY',base_url='http://0.0.0.0:30000/v1')
+model_name = client.models.list().data[0].id
+response = client.chat.completions.create(
+    model=model_name,
+    messages=messages,
+    temperature=0.8,
+    top_p=0.8,
+    stream=False,
+    tools=tools)
+print(response)
```

Comment:
1. (Necessary) Add a test case in https://github.com/sgl-project/sglang/blob/main/test/srt/test_openai_server.py. Assert the correctness of the output
2. (Optionally) convert this doc into an interactive notebook, similar to the others https://github.com/sgl-project/sglang/blob/main/docs/start/send_request.ipynb